cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -401,6 +403,55 @@ public synchronized void updateProperties(NamespaceId namespaceId, NamespaceMeta
       throw new BadRequestException(String.format("Mappings %s for namespace %s cannot be updated once the namespace " +
                                                     "is created.", difference, namespaceId));
     }
+
+    if (namespaceMeta.getRepoConfig() != null) {
+      updateRepoConfig(builder, namespaceMeta.getRepoConfig());
+    }
+    
+    NamespaceMeta updatedMeta = builder.build();
+    nsStore.update(updatedMeta);
+    // refresh the cache with new meta
+    namespaceMetaCache.refresh(namespaceId);
+    LOG.info("Namespace {} updated with meta {}", namespaceId, updatedMeta);
+  }
+
+  private void updateRepoConfig(NamespaceMeta.Builder builder, NamespaceRepositoryConfig repoConfig) {
+    if (repoConfig.getProvider() != null) {
+      builder.setRepoProvider(repoConfig.getProvider());
+    }
+    if (repoConfig.getLink() != null) {
+      builder.setRepoLink(repoConfig.getLink());
+    }
+    if (repoConfig.getDefaultBranch() != null) {
+      builder.setRepoDefaultBranch(repoConfig.getDefaultBranch());
+    }
+    if (repoConfig.getAuthType() != null) {
+      builder.setRepoAuthType(repoConfig.getAuthType());
+    }
+    if (repoConfig.getUsername() != null) {
+      builder.setRepoUserName(repoConfig.getUsername());
+    }
+    if (repoConfig.getPathPrefix() != null) {
+      builder.setRepoPathPrefix(repoConfig.getPathPrefix());
+    }
+  }
+
+  @Override
+  public synchronized void deleteRepository(NamespaceId namespaceId) throws Exception {
+    if (!exists(namespaceId)) {
+      throw new NamespaceNotFoundException(namespaceId);
+    }
+    accessEnforcer.enforce(namespaceId, authenticationContext.getPrincipal(), StandardPermission.UPDATE);
+
+    NamespaceMeta existingMeta = nsStore.get(namespaceId);
```

Comment:
Using store to do get and then update is not safe as the get and the update calls are happening in different transaction. This means it's possible to overwrite data unintentionally under concurrent calls. (e.g. call A updates the repo config, call B updates tethering config).